scripts/Stretch.py

```diff
@@ -0,0 +1,419 @@
+        If the triangle is not wide enough, there is a huge risk of finding
+        an incorrect orientation, due to insufficient accuracy.
+        So, when the consecutive points are too close, the method
+        use following and preceding points to form a wider triangle around
+        the current point
+        dmin_tri is the minimum distance between two consecutive points
+        of an acceptable triangle
+        """
+        dmin_tri = self.line_width / 2.0
+        iextra = np.floor_divide(len(orig_seq), 3) # Nb of extra points
+        ibeg = 0 # Index of first point of the triangle
+        iend = 0 # Index of the third point of the triangle
+        for i, step in enumerate(orig_seq):
+            # i is the index of the second point of the triangle
+            # pos_after is the array of positions of the original sequence
+            # after the current point
+            pos_after = np.resize(np.roll(orig_seq, -i-1, 0), (iextra, 2))
+            # Vector of distances between the current point and each following point
+            dist_from_point = ((step - pos_after) ** 2).sum(1)
+            if np.amax(dist_from_point) < dmin_tri * dmin_tri:
+                continue
+            iend = np.argmax(dist_from_point >= dmin_tri * dmin_tri)
+            # pos_before is the array of positions of the original sequence
+            # before the current point
+            pos_before = np.resize(np.roll(orig_seq, -i, 0)[::-1], (iextra, 2))
+            # This time, vector of distances between the current point and each preceding point
+            dist_from_point = ((step - pos_before) ** 2).sum(1)
+            if np.amax(dist_from_point) < dmin_tri * dmin_tri:
+                continue
+            ibeg = np.argmax(dist_from_point >= dmin_tri * dmin_tri)
+            # See https://github.com/electrocbd/post_stretch for explanations
+            # relpos is the relative position of the projection of the second point
+            # of the triangle on the segment from the first to the third point
+            # 0 means the position of the first point, 1 means the position of the third,
+            # intermediate values are positions between
+            length_base = ((pos_after[iend] - pos_before[ibeg]) ** 2).sum(0)
+            relpos = ((step - pos_before[ibeg])
+                      * (pos_after[iend] - pos_before[ibeg])).sum(0)
+            if np.fabs(relpos) < 1000.0 * np.fabs(length_base):
+                relpos /= length_base
+            else:
+                relpos = 0.5 # To avoid division by zero or precision loss
+            projection = (pos_before[ibeg] + relpos * (pos_after[iend] - pos_before[ibeg]))
+            dist_from_proj = np.sqrt(((projection - step) ** 2).sum(0))
+            if dist_from_proj > 0.001: # Move central point only if points are not aligned
+                modif_seq[i] = (step - (self.stretch / dist_from_proj)
+                                * (projection - step))
```

Comment:
The change on the triangle infill is not noticeable. I just tried. When the algorithm moves a point in a triangle, it is for a maximum distance of the stretch distance parameter, about 0.08 mm.